src/main/java/com/patchworkmc/ColorPane.java

```diff
@@ -0,0 +1,168 @@
+package com.patchworkmc;
+
+import java.awt.Color;
+import java.util.function.Supplier;
+
+import javax.swing.JTextPane;
+import javax.swing.text.BadLocationException;
+import javax.swing.text.Style;
+import javax.swing.text.StyleConstants;
+
+import org.apache.logging.log4j.Level;
+
+public class ColorPane extends JTextPane {
+	private static final Color D_Black = Color.getHSBColor(0.000f, 0.000f, 0.000f);
+	private static final Color D_Red = Color.getHSBColor(0.000f, 1.000f, 0.502f);
+	private static final Color D_Blue = Color.getHSBColor(0.667f, 1.000f, 0.502f);
+	private static final Color D_Magenta = Color.getHSBColor(0.833f, 1.000f, 0.502f);
+	private static final Color D_Green = Color.getHSBColor(0.333f, 1.000f, 0.502f);
+	private static final Color D_Yellow = Color.getHSBColor(0.167f, 1.000f, 0.502f);
+	private static final Color D_Cyan = Color.getHSBColor(0.500f, 1.000f, 0.502f);
+	private static final Color D_White = Color.getHSBColor(0.000f, 0.000f, 0.753f);
+	private static final Color B_Black = Color.getHSBColor(0.000f, 0.000f, 0.502f);
+	private static final Color B_Red = Color.getHSBColor(0.000f, 1.000f, 1.000f);
+	private static final Color B_Blue = Color.getHSBColor(0.667f, 1.000f, 1.000f);
+	private static final Color B_Magenta = Color.getHSBColor(0.833f, 1.000f, 1.000f);
+	private static final Color B_Green = Color.getHSBColor(0.333f, 1.000f, 1.000f);
+	private static final Color B_Yellow = Color.getHSBColor(0.167f, 1.000f, 1.000f);
+	private static final Color B_Cyan = Color.getHSBColor(0.500f, 1.000f, 1.000f);
+	private static final Color B_White = Color.getHSBColor(0.000f, 0.000f, 1.000f);
+
+	// TODO: instance-based and mutable when we have a dark mode
+	private static final boolean IS_LIGHT = true;
+	private static final Supplier<Color> cReset = () -> IS_LIGHT ? D_Black : D_White;
```

Comment:
Not a huge fan of this. It assumes that dark mode == full black which is often not true.